BUG: fix Series constructor for scalar and Categorical dtype

cbertinato opened the pull request:
Categories and ordering of the resulting Series were not the same as those of the passed Categorical dtype. Closes #19565

- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

TomAugspurger commented:
Could you also add a test that hits the `Series` constructor directly?

```python
result = Series('a', dtype=pd.api.types.CategoricalDtype(['a', 'b'], ordered=True)
```

That way if the implementation changes in the future, we at least have that covered.

jschendel commented on pandas/core/dtypes/cast.py:
```diff
@@ -1178,7 +1178,8 @@ def construct_1d_arraylike_from_scalar(value, length, dtype):
         from pandas import Categorical
-        subarr = Categorical([value] * length)
+        subarr = Categorical([value] * length, dtype.categories,
+                             ordered=dtype.ordered)
```
I think this could be done a bit more directly with `Categorical([value] * length, dtype=dtype)`.

cbertinato replied:
Will do. I'll add a test for the Series constructor. @jschendel good point. I'll make that change as well.

cbertinato committed:
BUG: fix Series constructor for scalar and Categorical dtype

Categories and ordering of the resulting Series were not the same as those of the passed Categorical dtype

cbertinato committed:
Fixed lint error

jreback commented:
thanks!
